src/commonMain/kotlin/StringUtils.kt

```diff
@@ -0,0 +1,5 @@
+internal val String.intValue: Int
+    get() = doubleValue.toInt()
+
+internal val String.doubleValue: Double
```

Comment:
I was checking how those values will behave, and we should not use here Double at all. We should move 100% to bigdecimal. Here is example case for quite big number, but that can happen to smaller ones

![image](https://user-images.githubusercontent.com/97684900/151553679-c7ab822a-a158-4879-a8aa-91649f80de80.png)

We will lose precision for sure, and as this can be used to count things there will be incorrect results. 

Not sure too if we prefer to override 'toInt' too. This was of doing makes sure we will get value, even if someone provides value with decimal parts, it will cut it and use. Having just kotlin .toInt makes "2.0" a 0, your code makes it 2. Question is, do we think it is correct or, if values are incorrect they shouldn't be 'fixed' by library. 

Those are I think topics to talk f2f on monday though. :)

Reply:
Big Decimal is a jvm specific solution and we cannot use it directly in the common implemenation.